src/components/BulkOperationDetailsPage/BulkOperationDetails.jsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import 'moment-timezone';
+import PropTypes from 'prop-types';
+import {
+  Button, DataTable, useToggle, FullscreenModal, Stack,
+} from '@openedx/paragon';
+
+import { formatDateTime } from '../../utils';
+import './styles.scss';
+
+const ImageCell = ({ value }) => (
+  value ? (
+    <a href={value} target="_blank" rel="noopener noreferrer">
+      View Image
+    </a>
+  ) : null
+);
+
+ImageCell.propTypes = {
+  value: PropTypes.string,
+};
+
+ImageCell.defaultProps = {
+  value: null,
+};
+
+const parseCSV = async (url) => {
+  const response = await fetch(url);
+  const text = await response.text();
+  const lines = text.trim().split('\n');
+
+  const headers = lines[0].split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+  const cleanedHeaders = headers.map((h) => h.replace(/(^"|"$)/g, ''));
+
+  const rows = lines.slice(1).map((line) => line.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/).map((cell) => cell.replace(/(^"|"$)/g, '')));
+
+  return { headers: cleanedHeaders, rows };
+};
+
+const BulkOperationDetails = ({ task }) => {
+  const [csvRows, setCsvRows] = useState([]);
+  const [csvHeaders, setCsvHeaders] = useState([]);
+  const [isOpen, open, close] = useToggle(false);
+  const DEFAULT_CSV_PREVIEW_PAGE_SIZE = 5;
+
+  if (!task || Object.keys(task).length === 0) {
+    return <p className="text-info"> No BulkOperationTask found against the given ID. Please check the ID and try again.</p>;
+  }
+
+  const handlePreviewCSV = async () => {
+    try {
+      const { headers, rows } = await parseCSV(task.csv_file);
+      setCsvHeaders(headers);
+      setCsvRows(rows);
+    } catch (error) {
+      console.error('Failed to preview CSV:', error);
```

Comment:
The error would need to be displayed to the user to show that the preview has gone wrong.